src/controller/TypedReadCallback.h

```diff
@@ -97,5 +97,57 @@ class TypedReadCallback final : public app::ReadClient::Callback
     OnDoneCallbackType mOnDone;
 };
 
+template <typename DecodableEventTypeInfo>
+class TypedReadEventCallback final : public app::ReadClient::Callback
+{
+public:
+    using OnSuccessCallbackType = std::function<void(const app::EventHeader & aEventHeader, const DecodableEventTypeInfo & aData)>;
+    using OnErrorCallbackType   = std::function<void(const app::EventHeader * apEventHeader,
+                                                   Protocols::InteractionModel::Status aIMStatus, CHIP_ERROR aError)>;
+    using OnDoneCallbackType    = std::function<void(app::ReadClient * client, TypedReadEventCallback * callback)>;
+
+    TypedReadEventCallback(ClusterId aClusterId, EventId aEventId, OnSuccessCallbackType aOnSuccess, OnErrorCallbackType aOnError,
+                           OnDoneCallbackType aOnDone) :
+        mClusterId(aClusterId),
+        mEventId(aEventId), mOnSuccess(aOnSuccess), mOnError(aOnError), mOnDone(aOnDone)
```

Comment:
Does not need mClusterId or mEventId, because those can be gotten from the template parameter.....  Notice that we don't use these members for anything!